### PR TITLE
[secure-store] Add `canUseBiometricAuthentication` function

### LINKED ISSUE #26777: Unable to determine Biometric Security Level
### Minimal reproducible example

https://snack.expo.dev/@prichmond/thankful-blue-mixed-nuts

### Summary

As per the details outlined in [this issue](https://github.com/expo/expo/issues/26615) I am aware that Android Keystore requires Strong Biometric authentication and that the current `isEnrolledAsync` function only checks for BIOMETRIC_WEAK (e.g. Face Recognition on most Android devices). 

If we are unable to determine the level of biometric enrolment before making calls to SecureStore which require strong biometrics, it makes it hard to prevent errors from occurring. 

Is it possible to update the existing `isEnrolledAsync` to allow a parameter to specify the level of biometrics you are checking, or create a new function to determine the user's current level? 

### Environment

expo-env-info 1.2.0 environment info:
System:
OS: macOS 14.2.1
Shell: 5.9 - /bin/zsh
Binaries:
Node: 18.18.0 - ~/Library/Caches/fnm_multishells/95796_1705881749006/bin/node
Yarn: 1.22.21 - /opt/homebrew/bin/yarn
npm: 9.8.1 - ~/Library/Caches/fnm_multishells/95796_1705881749006/bin/npm
Watchman: 2023.12.04.00 - /opt/homebrew/bin/watchman
Managers:
CocoaPods: 1.14.3 - /Users/peterrichmond/.rvm/gems/ruby-3.0.5/bin/pod
SDKs:
iOS SDK:
Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
IDEs:
Android Studio: 2021.1 AI-211.7628.21.2111.8193401
Xcode: 15.2/15C500b - /usr/bin/xcodebuild
npmPackages:
expo: ^48.0.21 => 48.0.21
react: 18.2.0 => 18.2.0
react-dom: 18.2.0 => 18.2.0
react-native: 0.71.14 => 0.71.14
react-native-web: ~0.18.12 => 0.18.12
Expo Workflow: managed

### PR DESCRIPTION
**Pull request by behenate**
# Why
fixes https://github.com/expo/expo/issues/26777
 
Secure-store will throw an error when trying to save a value with `requireAuthentication` set to `true` but no biometrics (or insufficient biometrics) are enrolled. This is good, but it's the only way (without using `expo-local-authentication`) of checking if the value can be saved. This is not ideal. 
This PR adds a small convenience function that allows our users to check if `requireAuthentication` can be used before saving the value.

Any suggestions for a better name for the method are appreciated! 😄 

# How

On Android used existing assertion function, handled the exception by returning `false` if it occurred, otherwise `true`. On iOS used `LAContext` to check currently enrolled biometrics.

# Test Plan

Tested in BareExpo on iOS 16 and Android 13 devices.

### CODE REVIEW

File: packages/expo-secure-store/android/src/main/java/expo/modules/securestore/SecureStoreModule.kt [hunk 1]
```diff
@@ -66,6 +66,15 @@ open class SecureStoreModule : Module() {
       }
     }
 
+    Function("canUseBiometricAuthentication") {
+      try {
+        authenticationHelper.assertBiometricsSupport()
+      } catch (e: AuthenticationException) {
+        return@Function false
+      }
+      return@Function true
```
Comment (lukmccall) [inline packages/expo-secure-store/android/src/main/java/expo/modules/securestore/SecureStoreModule.kt:75]:
```suggestion
      return@Function try {
        authenticationHelper.assertBiometricsSupport()
        true
      } catch (e: AuthenticationException) {
         false
      }
```

File: packages/expo-secure-store/ios/SecureStoreModule.swift [hunk 2]
```diff
@@ -49,6 +49,17 @@ public final class SecureStoreModule: Module {
...
+
+    Function("canUseBiometricAuthentication") {() -> Bool in
+      let context = LAContext()
+      var error: NSError?
+      let isBiometricsSupported: Bool = context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &error)
+
+      if isBiometricsSupported && error == nil {
+        return true
+      }
+      return false
```
Comment (lukmccall) [inline packages/expo-secure-store/ios/SecureStoreModule.swift:61]:
```suggestion
      if error != nil {
        return false
      }
      return isBiometricsSupported
```

**Review by lukmccall — APPROVED** [review]

File: packages/expo-secure-store/src/SecureStore.ts [hunk 3]
```diff
@@ -202,6 +202,14 @@ export function getItem(key: string, options: SecureStoreOptions = {}): string |
   return ExpoSecureStore.getValueWithKeySync(key, options);
 }
 
+/**
+ * Checks if the value can be saved with `requireAuthentication` option set to `true`.
```
Comment (amandeepmittal) [inline packages/expo-secure-store/src/SecureStore.ts:206]:
```suggestion
 * Checks if the value can be saved with `requireAuthentication` option enabled.
```

The last part of the sentence sounds confusing. We could either provide complete context that the "when that option is set to true" or just say "when that option is enabled" and be concise about it.

**Review by amandeepmittal — APPROVED** [review]
Lgtm 👍 

Left a minor suggestion.

**Review by tsapeta — APPROVED** [review]
Perhaps we could use the new function in NCL example and test-suite

**Pull request merged**